common/environment/environment.cpp

```diff
@@ -1723,6 +1726,21 @@ IConstInstanceInfo * CLocalEnvironment::getInstance(const char *type, const char
     return nullptr;
 }
 
+IConstInstanceInfoIterator * CLocalEnvironment::getInstanceIterator(const char *type, const char * name) const
+{
+    if (isEmptyString(type) || isEmptyString(name))
+        return nullptr;
+
+    StringBuffer xpath("Software/");
+    xpath.append(type);
+    if (name)
+        xpath.append("[@name='").append(name).append("']");
+    xpath.append("/Instance");
+
+    synchronized procedure(safeCache);
```

Comment:
Not convinced this is safe - you are returning an iterator out of this function, which will therefore NOT be protected by this mutex.

Reply:
I agree and spotted similar elsewhere when reviewing another PR, and opened this JIRA at the time: 
https://track.hpccsystems.com/browse/HPCC-22177